### PR TITLE
feature: enabled sending boolean and nil in cosocket.

### PR DESCRIPTION
**Pull request by spacewander**
I hereby granted the copyright of the changes in this pull request
to the authors of this lua-nginx-module project.

### CODE REVIEW

File: src/ngx_http_lua_socket_tcp.c [hunk 1]
```diff
@@ -2462,6 +2462,20 @@ ngx_http_lua_socket_tcp_send(lua_State *L)
...
+
+        case LUA_TBOOLEAN:
+            if (lua_toboolean(L, 2)) {
+                len = sizeof("true") - 1;
+
+            } else {
+                len = sizeof("false") - 1;
+            }
+
+            break;
```
Comment (agentzh) [inline src/ngx_http_lua_socket_tcp.c]: Sending booleans and nils are more likely due to bugs in the application code? I'm not sure if this is the right thing to do.

Comment (spacewander) [reply src/ngx_http_lua_socket_tcp.c]: @agentzh
It's for the sake of duck type. API like `ngx.log` and `ngx.say` handle boolean and nil by converting them to string. If we finally migrate cosocket to FFI implementation, we will use `tostring` to non-string type argument too. BTW, it also conforms the description in the error message: "string, number, boolean, nil, or array table expected".

Comment (agentzh) [reply src/ngx_http_lua_socket_tcp.c]: @spacewander OK then.

**Comment by agentzh** [comment]
We're migrating to meta-lua-nginx-module and we'll need existing PRs like this to be ported to that repo instead.

**Comment by agentzh** [comment]
@spacewander Merged. Thanks!

**Pull request closed without merging**